src/main/java/mate/academy/dao/BookDaoImpl.java

```diff
@@ -0,0 +1,116 @@
+package mate.academy.dao;
+
+import java.math.BigDecimal;
+import java.sql.PreparedStatement;
+import java.sql.ResultSet;
+import java.sql.SQLException;
+import java.sql.Statement;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Optional;
+import mate.academy.exception.DataProcessingException;
+import mate.academy.lib.Dao;
+import mate.academy.model.Book;
+import mate.academy.util.ConnectionUtil;
+
+@Dao
+public class BookDaoImpl implements BookDao {
+
+    public Book create(Book book) {
+        String sql = "INSERT INTO books (title, price) VALUES (?, ?)";
+
+        try (PreparedStatement statement = ConnectionUtil.getConnection().prepareStatement(sql,
+                Statement.RETURN_GENERATED_KEYS)) {
+            statement.setString(1, book.getTitle());
+            statement.setBigDecimal(2, book.getPrice());
+
+            int affectedRows = statement.executeUpdate();
+
+            if (affectedRows == 0) {
+                throw new SQLException("Creating book failed, no rows affected.");
+            }
+
+            try (ResultSet generatedKeys = statement.getGeneratedKeys()) {
+                if (generatedKeys.next()) {
+                    book.setId(generatedKeys.getLong(1));
+                } else {
+                    throw new SQLException("Creating book failed, no ID obtained.");
+                }
+            }
+        } catch (SQLException e) {
+            throw new DataProcessingException("Can't create book " + book, e);
+        }
+
+        return book;
+    }
+
+    public Optional<Book> findById(Long id) {
+        String sql = "SELECT * FROM books WHERE id = ?";
+
+        try (PreparedStatement statement = ConnectionUtil.getConnection().prepareStatement(sql)) {
+            statement.setLong(1, id);
+
+            try (ResultSet resultSet = statement.executeQuery()) {
+                if (resultSet.next()) {
+                    Book book = new Book();
+                    book.setId(resultSet.getObject("id", Long.class));
+                    book.setPrice(resultSet.getObject("price", BigDecimal.class));
+                    book.setTitle(resultSet.getString("title"));
```

Comment:
Avoid using both setters and constructors for initializing objects. Choose one approach for consistency. Also, consider moving ResultSet processing into a separate private method to avoid code duplication.